pkg/controller/controller.go

```diff
@@ -594,7 +613,21 @@ func (h *Handler) syncProvisionerDaemonset(cr *localv1.LocalVolume, forceRollout
 	ds := h.generateLocalProvisionerDaemonset(cr)
 	dsName := ds.Name
 	generation := getExpectedGeneration(cr, ds)
-	ds, _, err := h.apiClient.applyDaemonSet(ds, generation, forceRollout)
+
+	// See if there's an existing DaemonSet in the cluster. If so,
+	// hash in the annotations to determine if we need to force
+	// an update.
+	existingDS, err := h.apiClient.getDaemonSet(ds.Namespace, dsName)
+	if err != nil {
+		klog.Info("Error getting existing provisioner DaemonSet: %v", err)
+	}
+	existingAnnotations := existingDS.ObjectMeta.Annotations
+	if existingAnnotations == nil ||
+		existingAnnotations[specHashAnnotation] != ds.ObjectMeta.Annotations[specHashAnnotation] {
+		forceRollout = true
+	}
+
```

Comment:
Have we tried pulling latest version of library-go which automatically does this - https://github.com/openshift/aws-ebs-csi-driver-operator/pull/49/files ?

Reply:
Also if we can't pull library-go then we should at least move this code that duplicates bunch of code in a function of its own and then write some tests.

Reply:
As discussed offline - there are build issues after updating to the latest library-go. I've created a method that does this check.